master/internal/api_trials.go

```diff
@@ -947,9 +947,13 @@ func (a *apiServer) AllocationPreemptionSignal(
 
 	id := uuid.New()
 	var w task.PreemptionWatcher
-	if err := a.ask(handler.Address(), task.WatchPreemption{
+	err = a.ask(handler.Address(), task.WatchPreemption{
 		ID: id, AllocationID: allocationID,
-	}, &w); err != nil {
+	}, &w)
+
+	if _, ok := err.(task.ErrAllocationUnfulfilled); ok {
+		return nil, status.Errorf(codes.Unavailable, "allocation isn't active yet")
```

Comment:
does this cause the job to fail? seems like we would want to know we are in a restoring state and to try again soon.